sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -151,20 +151,36 @@ private void log(int logLevel, String format, Object... args) {
     }
 
     /**
-     * This will log {@link RuntimeException}, if logging is enabled , and throw the runtime exception.
-     * @param runtimeException to be thrown. It will do nothing if {@code null} is provided.
-     * @throws RuntimeException which is requested by this call.
+     * Attempts to log the exception at the warning level and returns it to be thrown.
+     *
+     * @param runtimeException Runtime exception to be logged and returned.
+     * @return the passed {@code RuntimeException}
+     */
+    public RuntimeException logWarningAndThrow(RuntimeException runtimeException) {
+        return logAndThrow(runtimeException, WARNING_LEVEL);
+    }
+
+    /**
+     * Attempts to log the exception at the error level and returns it to be thrown.
+     *
+     * @param runtimeException Runtime exception to be logged and returned.
+     * @return the passed {@code RuntimeException}
      */
-    public void logAndThrow(RuntimeException runtimeException) {
+    public RuntimeException logErrorAndThrow(RuntimeException runtimeException) {
```

Comment:
What scenario does logWarningAndThrow cover that logErrorAndThrow doesn't? They both return exceptions, which indicate errors. I may be missing context.

Reply:
Looking at the [logging guidelines](https://azure.github.io/azure-sdk/java_implementation.html#logging), since these are RuntimeExceptions being thrown, it feels fatal. 
The program's probably not going to recover from it. And in the usages below where we log warning and throw in the constructor for an IllegalArgumentException, it seems like a fatal error to me. 🤔

Reply:
The changed calls may not be the best for why `logWarningAndThrow`, or whatever it gets named, is needed. There are cases where a CRUD call may throw an exception but it should be recoverable, such as a message failing to send or an upload dying due to a network hiccup.